### PR TITLE
[FCE-603] Karkakol/add observer pattern

### PR DESCRIPTION
**Pull request by karkakol**
Utile observer pattern

### CODE REVIEW

File: packages/react-native-client/ios/managers/LocalCameraTracksChangedListenersManager.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class LocalCameraTracksChangedListenersManager {
+    private var listeners: [LocalCameraTrackChangedListener] = []
+
+    func add(_ listener: LocalCameraTrackChangedListener) {
+        listeners.append(listener)
+    }
+
+    func remove(_ listener: LocalCameraTrackChangedListener) {
+      listeners.removeAll{ $0 === listener}
+
```
Comment (maksg) [inline packages/react-native-client/ios/managers/LocalCameraTracksChangedListenersManager.swift:16]: Empty line

File: packages/react-native-client/ios/managers/TracksUpdateListenersManager.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class TracksUpdateListenersManager {
+    private var listeners: [TrackUpdateListener] = []
+
+    func add(_ listener: TrackUpdateListener) {
+        listeners.append(listener)
+    }
+
+    func remove(_ listener: TrackUpdateListener) {
+      listeners.removeAll{$0 === listener}
```
Comment (maksg) [inline packages/react-native-client/ios/managers/TracksUpdateListenersManager.swift:13]: Same as above

File: packages/react-native-client/ios/managers/LocalCameraTracksChangedListenersManager.swift (hunk 1 above)
Comment (karkakol) [reply packages/react-native-client/ios/managers/LocalCameraTracksChangedListenersManager.swift]: Some lint issues, i fixed it.

**Review by maksg — APPROVED** [review]

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/VideoPreviewViewModule.kt [hunk 3]
```diff
@@ -9,6 +9,11 @@ class VideoPreviewViewModule : Module() {
       Name("VideoPreviewViewModule")
 
       View(VideoPreviewView::class) {
+        OnViewDestroys { view: VideoPreviewView ->
+          view.dispose()
+          RNFishjamClient.localTracksSwitchListenerManager.remove(view)
```
Comment (MiloszFilimowski) [inline packages/react-native-client/android/src/main/java/io/fishjam/reactnative/VideoPreviewViewModule.kt:14]: **question:** Why is this needed? `onDetachedFromWindow` is not enough?

**Review by MiloszFilimowski — APPROVED** [review]
Great work!

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/VideoPreviewViewModule.kt (hunk 3 above)
Comment (karkakol) [reply packages/react-native-client/android/src/main/java/io/fishjam/reactnative/VideoPreviewViewModule.kt:14]: I wil explain irl.
